### PR TITLE
Fix some list functions

### PR DESCRIPTION
**Pull request by sjoelund**

### CODE REVIEW

File: src/list.jl [hunk 1]
```diff
@@ -109,7 +113,7 @@ List(T::Type #= Hack.. =#, args) = let
   end
   local lst1::Cons{T} = Cons{T}(convert(T, args.head) ,nil)
   if args.tail isa Nil
-    return nil
+    return lst1
```
Comment (JKRT) [inline src/list.jl]: We need to do a conversion here I think so we should return Cons{T}(convert(T, lst1.head), nil)

Comment (JKRT) [reply src/list.jl]: nvm

File: test/listTests.jl [hunk 2]
```diff
@@ -206,7 +219,8 @@ end
   end
   @test true == let
       try
-        a::List{List{Integer}} = list(list(1))
+        # TODO: Fix this to allow Integer
+        a::List{List{Int64}} = list(list(1))
```
Comment (JKRT) [inline test/listTests.jl]: Hmm this needs to work otherwise the Julia frontend will not compile

Comment (sjoelund) [reply test/listTests.jl]: Agreed. Something must be missing for this.

Comment (JKRT) [reply test/listTests.jl]: @sjoelund  I just made some changes. It works with the tests you added. Could you verify that it still works for all cases?

**Pull request merged**